cmd/collector/main.go

```diff
@@ -90,20 +90,21 @@ func main() {
 			builderOpts := new(builder.CollectorOptions).InitFromViper(v)
 
 			mBldr := new(pMetrics.Builder).InitFromViper(v)
-			metricsFactory, err := mBldr.CreateMetricsFactory("jaeger-collector")
+			baseFactory, err := mBldr.CreateMetricsFactory("jaeger")
```

Comment:
why are we removing the `-collector` suffix here? Isn't it better to be able to tell apart which binary emitted the metric? The storage metrics won't be grouped by collector name anymore.

Reply:
> why are we removing the -collector suffix here?

It's just so that the `jaeger` prefix is used for the storage metrics without the component's name. `collector` is added later as a namespace to the collector-specific metrics.

> Isn't it better to be able to tell apart which binary emitted the metric?

I think the agreement was to have per-component metrics, instead of per-binary, so that metrics generated by the collector via `standalone-linux` would be consistent with the metrics generated by collector via `collector-linux`:

(before)
`jaeger_standalone_jaeger_collector_in_queue_latency_bucket{host="caju",le="0.005"} 0`

vs

(after, across all binaries)
`jaeger_collector_in_queue_latency_bucket{host="caju",le="0.005"} 0`

> The storage metrics won't be grouped by collector name anymore.

You mean, `jaeger_cassandra_inserts{table="service_names"}` would be used by `collector`, `query` and `standalone`? This would potentially create three different metric names for the same thing: `jaeger_standalone_cassandra_inserts`, `jaeger_collector_cassandra_inserts`, `jaeger_query_cassandra_inserts`. I personally prefer to have one single metrics, but I do see the point of having one per *component* (not binary). Just not sure how we'd tell the components apart when using the standalone.

Reply:
If knowing the source binary is important, then it could be added as a label instead?

Reply:
I am really surprised Prometheus community has no guidelines around this. Can't imagine any minimally complex system not running into the same exact issues with multiple namespaces and hierarchies.

Reply:
> I am really surprised Prometheus community has no guidelines around this. Can't imagine any minimally complex system not running into the same exact issues with multiple namespaces and hierarchies.

The Prometheus scraper can add arbitrary labels, so, this kind of information would arguably be added there, along with some metadata not known to the "microservice", like the Geo, DC, Rack, and so on. Our "binary" name could also be added at scrape time, for cases where it has more value than a consistent set of metrics across all binaries.

https://prometheus.io/docs/prometheus/latest/configuration/configuration/#%3Cstatic_config%3E

How about we merge it with the current state, publish a blog post about this change + how to add arbitrary labels, and ask for community feedback? If we hear that having this as part of the metrics we emit, we add it as a label.

Reply:
Maybe we should discuss on Friday. We are accumulating a set of breaking changes for 1.5: metrics, docker commands, and Cassandra schema.